src/models/embeddingsList.ts

```diff
@@ -0,0 +1,135 @@
+    Object.setPrototypeOf(this, EmbeddingsList.prototype);
+    this.model = model;
+    this.data = data;
+    this.usage = usage;
+  }
+
+  /* Customize format of output. */
+  public toString(): string {
+    const truncatedData = this.truncateData();
+    const dataObject = truncatedData
+      .map((embedding) => {
+        if (typeof embedding === 'string') {
+          return `    ${embedding}`;
+        }
+        let embeddingObject = JSON.stringify(embedding, (key, value) =>
+          key === 'values' && Array.isArray(value) ? value : value
+        );
+        embeddingObject = embeddingObject.replace(/:/g, ': ');
+
+        // Format the embedding itself
+        const valuesArray =
+          embeddingObject.match(/"values": \[(.*?)\]/)?.[1] || '';
+        const formattedEmbedding = valuesArray
+          .split(',')
+          .join(', ')
+          .replace(/"/g, '');
+
+        // Replace the right side of the colon after "values: "
+        embeddingObject = embeddingObject.replace(
+          /("values": )\[(.*?)\]/,
+          `$1[${formattedEmbedding}]`
+        );
+
+        return `    ${embeddingObject}`;
+      })
+      .join(',\n');
+
+    const usageObject = JSON.stringify(this.usage).replace(/:/g, ': ');
+    return (
+      `EmbeddingsList({\n` +
+      `  "model": "${this.model}",\n` +
+      `  "data": [\n` +
+      `${dataObject}\n` +
+      `   ],\n` +
+      `  "usage": ${usageObject}\n` +
+      `  })`
+    );
+  }
+
+  // Customize console output
+  [util.inspect.custom](): string {
+    return this.toString();
+  }
+
+  public toJSON(): any {
+    return {
+      model: this.model,
+      data: this.truncateData(),
+      usage: this.usage,
+    };
+  }
+
+  public get(index: number): Embedding {
+    return this[index];
+  }
+
+  public indexOf(element: Embedding): number {
+    return this.data ? this.data.indexOf(element) : -1;
+  }
+
+  /* Truncate the content of an embedding in the output when there are >5 numbers. */
+  truncateValues(values: number[]): any[] {
+    if (!values || values.length <= 4) {
+      return values ? values : [];
+    }
+    return [...values.slice(0, 2), '...', ...values.slice(-2)];
+  }
+
+  /* Truncate the number of embedding objects in the output when there are more >6 embeddings. */
+  truncateData(): Array<any> {
+    if (!this.data) return [];
+    if (this.data.length <= 5) {
+      return this.data.map((embedding) => ({
+        values: embedding.values ? this.truncateValues(embedding.values) : [],
+      }));
+    }
+    return [
+      ...this.data.slice(0, 2).map((embedding) => ({
+        values: embedding.values ? this.truncateValues(embedding.values) : [],
+      })),
+      `   ... (${this.data.length - 4} more embeddings) ...`,
+      ...this.data.slice(-2).map((embedding) => ({
+        values: embedding.values ? this.truncateValues(embedding.values) : [],
+      })),
+    ];
+  }
+}
```

Comment:
Can we rename these function to something like `truncateXXXForDisplay`? Otherwise I think it confuse someone about their intent

Reply:
Sure!